packages/reputation-miner/ReputationMiner.js

```diff
@@ -1313,67 +1346,59 @@ class ReputationMiner {
-       AND users.address!="0x0000000000000000000000000000000000000000"
-       ORDER BY reputations.value DESC`
-    );
-    await db.close();
+    const res = await this.queries.getAddressesWithReputation.all(reputationRootHash, colonyAddress.toLowerCase(), skillId);
     const addresses = res.map(x => x.user_address)
     return addresses;
   }
 
+  async getReputationsForAddress(reputationRootHash, colonyAddress, userAddress) {
+    const res = await this.queries.getReputationsForAddress.all(reputationRootHash, colonyAddress.toLowerCase(), userAddress.toLowerCase());
+    return res.map(function(x){ return {
+      skill_id: x.skill_id,
+      reputation: `0x${x.value.slice(2,66)}`
+    }});
+  }
+
   async createDB() {
-    const db = await sqlite.open({filename: this.dbPath, driver: sqlite3.Database});
-    await db.run("CREATE TABLE IF NOT EXISTS users ( address text NOT NULL UNIQUE )");
-    await db.run("CREATE TABLE IF NOT EXISTS reputation_states ( root_hash text NOT NULL UNIQUE, n_leaves INTEGER NOT NULL)");
-    await db.run("CREATE TABLE IF NOT EXISTS colonies ( address text NOT NULL UNIQUE )");
-    await db.run("CREATE TABLE IF NOT EXISTS skills ( skill_id INTEGER PRIMARY KEY )");
-    await db.run(
+    // Not regularly used, so not preparing them and saving the statement for reuse
+    await this.db.prepare("CREATE TABLE IF NOT EXISTS users ( address text NOT NULL UNIQUE )").run();
+    await this.db.prepare("CREATE TABLE IF NOT EXISTS reputation_states ( root_hash text NOT NULL UNIQUE, n_leaves INTEGER NOT NULL)").run();
+    await this.db.prepare("CREATE TABLE IF NOT EXISTS colonies ( address text NOT NULL UNIQUE )").run();
+    await this.db.prepare("CREATE TABLE IF NOT EXISTS skills ( skill_id INTEGER PRIMARY KEY )").run();
+    await this.db.prepare(
       `CREATE TABLE IF NOT EXISTS reputations (
         reputation_rowid text NOT NULL,
         colony_rowid INTEGER NOT NULL,
         skill_id INTEGER NOT NULL,
         user_rowid INTEGER NOT NULL,
         value text NOT NULL
       )`
-    );
+    ).run();
 
     // Do we have to do a database upgrade, from when we renamed n_nodes to n_leaves?
-    const nNodesColumn = await db.all("SELECT * FROM PRAGMA_TABLE_INFO('reputation_states') WHERE name='n_nodes';")
+    const nNodesColumn = await this.db.prepare("SELECT * FROM PRAGMA_TABLE_INFO('reputation_states') WHERE name='n_nodes';").all()
     if (nNodesColumn.length === 1) {
-      await db.run("ALTER TABLE 'reputation_states' RENAME COLUMN n_nodes to n_leaves");
-      const check1 = await db.all("SELECT * FROM PRAGMA_TABLE_INFO('reputation_states') where name='n_nodes'")
-      const check2 = await db.all("SELECT * FROM PRAGMA_TABLE_INFO('reputation_states') where name='n_leaves'")
+      await this.db.prepare("ALTER TABLE 'reputation_states' RENAME COLUMN n_nodes to n_leaves").run();
+      const check1 = await this.db.prepare("SELECT * FROM PRAGMA_TABLE_INFO('reputation_states') where name='n_nodes'").all()
```

Comment:
What's the difference between the `run` and `all` functionality? Is the latter a select?

Reply:
`run` executes the statement, but doesn't expect data back from the statement. `all` returns all rows in a `SELECT` statement, and `get` would return the first row only.